### PR TITLE
Downtime web form

### PR DESCRIPTION
**Pull request by matyasselmeci**
This is a web form for generating YAML blocks to be pasted into a downtime YAML file. It's running on topology-itb: https://topology-itb.opensciencegrid.org/

### CODE REVIEW

**Comment by matyasselmeci** [comment]
This includes the code reorganization and bug fix from #167 which we may want to merge sooner.

File: src/app.py [hunk 1]
```diff
@@ -186,27 +86,98 @@ def schema(xsdfile):
...
+or <a href="/{path}">select another facility.</a>
+""", 400))
+
+    if not form.validate_on_submit():
+        return render_template(template, form=form, resource=resource, facility=facility)
 
-    authorized = _get_authorized()
-    vos_xml = to_xml_bytes(global_data.get_vos_data().get_tree(authorized, filters))
-    return Response(vos_xml, mimetype='text/xml')
+    filename = "topology/" + topo.downtime_path_by_resource[resource]
```
Comment (brianhlin) [inline src/app.py]: Can we make this a little smarter? 

- If the file exists, let's link them directly to the edit form, e.g. https://github.com/opensciencegrid/topology/edit/master/topology/University%20of%20Wisconsin/GLOW/GLOW_downtime.yaml.
- If the file doesn't exist:
    1. Link them to the relevant site dir, e.g. https://github.com/opensciencegrid/topology/tree/master/topology/University%20of%20Wisconsin/GLOW
   1. Tell them to click the "Create new file" and give them the appropriate downtime name

File: src/webapp/forms.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        ("Severe", "Severe"),
+        ("Intermittent Outage", "Intermittent Outage"),
+        ("No Significant Outage Expected", "No Significant Outage Expected")
+    ])
+    description = StringField("Description", [InputRequired()])
+    start_date = DateField("Start Date", [InputRequired()])
+    start_time = TimeField("Start Time", [InputRequired()])
+    end_date = DateField("End Date", [InputRequired()])
+    end_time = TimeField("End Time", [InputRequired()])
+    services = SelectMultipleField("Services (select one or more)", [InputRequired()], choices=[])
```
Comment (brianhlin) [inline src/webapp/forms.py]: Can we give these fields tooltips?

File: src/webapp/forms.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        csrf = False  # CSRF not needed because no data gets modified
+
+    # https://stackoverflow.com/a/21815180
+    def validate(self):
+        if not super().validate():
+            return False
+        start_datetime = datetime.datetime.combine(self.start_date.data, self.start_time.data)
+        end_datetime = datetime.datetime.combine(self.end_date.data, self.end_time.data)
+        if start_datetime >= end_datetime:
+            self.end_time.errors.append("End date/time must be after start date/time")
```
Comment (brianhlin) [inline src/webapp/forms.py]: Can we put any input error messages on the right, in the same area that the yaml would get generated? Also, making them an angry red would make them more readily visible

Comment (matyasselmeci) [reply src/webapp/forms.py]: That would be pretty hard to do -- the error messages are formatted by the framework. I can do the angry red though.

File: src/webapp/forms.py (hunk 2 above)
Comment (matyasselmeci) [reply src/webapp/forms.py]: That looks like it would take a lot of code.

File: src/templates/generate_downtime_form.html [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+<br/>
+<form method=post>
+    <!-- NOTE: we are not using CSRF because none of the POST data is actually
+    used to change anything.  If that ever changes, see
+    https://flask-wtf.readthedocs.io/en/stable/csrf.html
+    -->
+    {{ render_hidden_field(form.resource) }}
+    <table>
+        {{ render_field(form.scheduled) }}
+        {{ render_fieldinfo("Outage known about for at least 48 hours in advance.") }}
```
Comment (brianhlin) [inline src/templates/generate_downtime_form.html]: Outages registered at least 48 hours in advance

File: src/templates/generate_downtime_form.html [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        {{ render_field(form.severity) }}
+        {{ render_fieldinfo("How much of the resource the outage affects.") }}
```
Comment (brianhlin) [inline src/templates/generate_downtime_form.html]: How much of the resource is affected

File: src/templates/generate_downtime_form.html [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            </td>
+        </tr>
+    </table>
+</form>
+</div>
+    <div class="right">
+        {% if yaml %}
+            {% if site_dir_url %}
+                {% if edit_url %}
+                    <a href="{{ edit_url }}" target="_blank">
```
Comment (brianhlin) [inline src/templates/generate_downtime_form.html]: Can we make this an ordered list?

**Comment by matyasselmeci** [comment]
OK, updated. Made a few other tweaks.

**Pull request merged**